pkg/virtctl/hotplug/cpu.go

```diff
@@ -0,0 +1,113 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Contributors
+ *
+ */
+
+package hotplug
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/clientcmd"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+var cpuCount uint32
+
+func NewCPUCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "cpu (VMI)",
+		Short:   "Set the number of CPUs for a running virtual machine instance.",
+		Example: cpuUsage(),
+		Args:    templates.ExactArgs("cpu", 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := CPU{clientConfig: clientConfig}
+			return c.Run(args)
+		},
+	}
+
+	cmd.Flags().Uint32Var(&cpuCount, "count", 0, "Number of CPUs to set.")
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+type CPU struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+func cpuUsage() string {
+	usage := `  # Set the number of CPUs to 2 for VirtualMachine 'myvm':
+  {{ProgramName}} hotplug cpu --count=2 myvm`
+
+	return usage
+}
+
+func (c *CPU) Run(args []string) error {
+	namespace, _, err := c.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	vmName := args[0]
+
+	if cpuCount == 0 {
+		return fmt.Errorf("cpuCount parameter must be positive")
+	}
+
+	virtCli, err := kubecli.GetKubevirtClientFromClientConfig(c.clientConfig)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), RequestTimeout)
```

Comment:
I like the idea of a request timeout. Perhaps we could add it as a common flag so other commands could make use of it as well?

Reply:
I would love to add that, but in a separate PR, since it's a bit out of scope :)